datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -2364,6 +2368,7 @@ public void run() {
           CoordinatorEvent event = _eventQueue.take();
           if (event != null) {
             handleEvent(event);
+            _isLeaderBusy = false;
```

Comment:
nit: should we set and unset _isLeaderBusy in the same scope ? Much easier to maintain the code that way. 

One way is:

_isLeaderBusy = _adapter.isLeader();
handleEvent();
_isLeaderBusy = false;

OR set and unset inside handleEvent() itself. Wdyt ?

Reply:
I agree with you that it would be better to set and unset within the same scope. I think I will go with setting/unsetting inside the run() method as that is what the leader coordinator thread scope is. Right now we do `handleEvent()` in the future, if we enhance leader to perform other tasks, then we do not need to worry about changing anything.